libraries/helper.rb

```diff
@@ -23,4 +23,31 @@ def entity_uuid
       return Chef::DataCollector::Messages.node_uuid
     end
   end
+
+  def get_tests_for_runner
+    profiles = node['audit']['profiles']
+    profiles_array = []
+    profiles.each { |profile| profiles_array.push(profile[:url] || profile[:git] || profile[:source]) }
+
+    profiles.each { |profile|
+      if profile.length == 1 && profile[:name]
+        profiles_array.push("supermarket://" + profile[:name])
```

Comment:
we would need to adapt the `add_target` method in inspec: https://github.com/chef/inspec/blob/master/lib/inspec/runner.rb#L156-L164 to receive objects. By default it is using that function to change from url string to hash object: https://github.com/chef/inspec/blob/master/lib/inspec/profile.rb#L43-L46.

Instead we would invoke https://github.com/chef/inspec/blob/master/lib/inspec/profile.rb#L38-L41 from the runner initially. See how it is invoked at https://github.com/chef/inspec/blob/master/lib/inspec/dependencies/requirement.rb#L101-L106